src/core/hw/gfxip/gfx9/gfx9CmdUtil.cpp

```diff
@@ -3270,6 +3270,8 @@ size_t CmdUtil::BuildSetPredication(
 
     // The predication memory address must be 16-byte aligned, and cannot be wider than 40 bits.
     PAL_ASSERT(((gpuVirtAddr & 0xF) == 0) && (gpuVirtAddr <= ((1uLL << 40) - 1)));
+    // The predicate type has to be valid.
+    PAL_ASSERT(predType < PredicateType::Boolean32);
```

Comment:
This assert only should only apply if memory32bPredicationSupport==0.

Reply:
I won't update this assert in this patch because `CmdUtil::BuildSetPredication` itself does not support a 32-bit predicate as of right now. When it is updated with a new `PFP_SET_PREDICATION_pred_op_enum` for 32-bit predicates and its logic is updated to support it, then the assert should be changed.

Reply:
I'll pull in a new copy of the PFP headers and remove the assert in a follow up change then. Heads up that the new enum value will be `pred_op__pfp_set_predication__Vulkan = 4`, so it should all work transparently in the existing code paths.